Overhauling config

Nischhelm opened the pull request:
added a lot of //TODO: nischcomment where i wasn't sure what is intended

absolutely not done, just wanted you to be able to see it already

Nischhelm committed:
initial clean

moved configs into their own folders
moved lazy list/set into configprovider and cleaned the code

Nischhelm committed:
more movement

Nischhelm committed:
minor features comments

Nischhelm committed:
integration config

Nischhelm committed:
server config

mainly moving stuff

Nischhelm committed:
loot config not done

cdstk commented:
Notes for myself about the todos:

ScaleWithMobLevels.java 57 - Discord dms, split into 1. actual scale func + 2. simplify towards vanilla looting func
LootConfig.java 63 - Remove configs, relic of rlmixins copy paste.
EntityLootHandler.java 50 - Fix key removal, supposed to act and be structured like the emerald/book drop removal.
ItemStaffSummoningElementLevelMapMixin.java 60 - Remove toLower, inefficient vanilla lycanites copy paste. JSON defaults already fit lowercase convention.

Nischhelm committed:
lootconfig done

Nischhelm committed:
potion config

also tweaked the event handler
and made the bosses not use the effects if they are not registered

Nischhelm committed:
quick import optimisation

expected more
dont ask me why the jsons get reordered from that oml

Nischhelm committed:
soulkey config done

Nischhelm committed:
loot funcs/conds

Nischhelm committed:
streamified

allowed all string list configs to have arbitrary amount of whitespace using .trim()
moved all config changed event stuff to main config class, rest via subfolder.reset()

cdstk commented:
Notes from running https://github.com/cdstk/LycanitesTweaks/pull/2/commits/127af6d67581723151b381bb3183f1b28a2de5b8

Run time null pointer in ModLoadedUtil from rlcombat check. Fixed in dev branch
Hard crash when voided/consumed handles a damage source with ImmediateSource is null (Fix known, check discord dms. Plan to use unique damage type string)
Did not do a server test, todo after merge

I plan to merge in a few days. This PR first, and then my dev branch.
After the merges, Configs will be prefixed with this numbering system. 
[n.c] Where n is a number starting from 0 to form config groups, and c is an optional letter used to set order within the group. Sole example from main https://github.com/cdstk/LycanitesTweaks/blob/05cfb991ef483cce928b74b655b505ba21f3407c/src/main/java/lycanitestweaks/handlers/config/CreatureStatsConfig.java

Nischhelm committed:
fixed dmg source copying
